src/samVisualize/commands/visualizeSamTemplate/samVisualizationManager.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { getLogger, Logger } from '../../../shared/logger/logger'
+import * as vscode from 'vscode'
+import { SamVisualization } from './samVisualization'
+export class SamVisualizationManager {
+    private readonly extensionContext: vscode.ExtensionContext
+    // Maps a path to a template file to a SamVisualization
+    public readonly managedVisualizations: Map<string, SamVisualization> = new Map<string, SamVisualization>()
+
+    constructor(extensionContext: vscode.ExtensionContext) {
+        this.extensionContext = extensionContext
+    }
+
+    /**
+     * Creates and renders a new SamVisualization, or shows an existing one.
+     * @param activeTextEditor The TextEditor whose contents are used to construct a SamVisualization
+     * @returns
+     */
+    public renderSamVisualization(activeTextEditor: vscode.TextEditor | undefined): vscode.WebviewPanel | undefined {
+        const logger: Logger = getLogger()
+
+        // Output channel is considered a text editor by VSCode (which we don't want)
+        // This check is required to prevent integration tests from failing due to the Go extension
+        if (!activeTextEditor || activeTextEditor.document.fileName.includes('extension-output')) {
+            logger.error('Could not get active text editor for SAM Template render.')
```

Comment:
Any other information you can include in this message?

Reply:
Something like `"Ensure there is a text editor open before attempting to render a template"`?